packages/@cdktf/hcl2cdk/lib/expressions.ts

```diff
@@ -122,6 +125,21 @@ function expressionForSerialStringConcatenation(nodes: t.Expression[]) {
         return node;
       }
 
+      // wrap access to dynamic blocks in Token.asString() as they return a Lazy
+      // for .key and .value which can't be concatenated in languages like Python
+      // because JSII currently has no support for the toString() method in
+      // languages other than TypeScript: https://github.com/aws/jsii/issues/380
+      // example: dynamic_iterator0.key / dynamic_iterator0.value
+      if (
+        t.isMemberExpression(node) &&
+        t.isIdentifier(node.object) &&
+        Object.values(scope.scopedVariables || {}).includes(node.object.name) &&
+        t.isIdentifier(node.property) &&
+        ["key", "value"].includes(node.property.name)
+      ) {
+        node = template.expression(`Token.asString(%%expr%%)`)({ expr: node });
```

Comment:
Can we make this a call to `coerceType` with the target type being string and the source type being dynamic?